airflow/utils/log/s3_task_handler.py

```diff
@@ -124,7 +124,8 @@ def s3_log_exists(self, remote_log_location):
         try:
             return self.hook.get_key(remote_log_location) is not None
         except Exception:
-            pass
+            msg = 'Could not list logs at {}'.format(remote_log_location)
+            self.log.exception(msg)
```

Comment:
My two cents:

1. maybe not necessary to have this intermediary variable `msg` here?
2. This function is supposed to return `True`/`False` in whatever situation, instead of raising exception. Raising an exception here is breaking the logic. This is why the tests are failing.